packages/uicore/src/components/MultiTypeInput/MultiTypeInput.tsx

```diff
@@ -201,6 +201,45 @@ export function ExpressionAndRuntimeType<T = unknown>(props: ExpressionAndRuntim
     }
   }, [multitypeInputValue])
 
+  const renderButtonType = useCallback(() => {
+    if (!allowableTypes.length) return null
+
+    return (
+      <Button
```

Comment:
would making this pure and moving it out of the MultiTypeInput component have the same effect? that might be a cleaner approach.

Reply:
Yes, thanks, updating